sound/soc/intel/boards/sof_da7219_max98373.c

```diff
@@ -146,17 +156,24 @@ static int da7219_codec_init(struct snd_soc_pcm_runtime *rtd)
 
 static int speaker_amp_init(struct snd_soc_pcm_runtime *rtd)
 {
-	int ret;
+	const struct snd_soc_dapm_widget *snd_widget = max98373_widgets;
+	const struct snd_soc_dapm_route *snd_route = max98373_map;
+	int ret, size = 2;
+
+	/* Note card name has sof- prefix */
+	if (!strcmp(rtd->card->name, "sof-da7219max98360a")) {
+		snd_widget = max98360a_widgets;
+		snd_route = max98360a_map;
+		size = 1;
```

Comment:
This reminds me of something... I've just recently commented on a similar pattern in a different (I think) PR. No need to do
```
int x = 1;

if (A)
	x = 2;
```
Seems much more logical to me to do
```
int x;

if (A)
	x = 2;
else
	x = 1;
```
for all the three variables above of course. But if you do 2 functions, this becomes irrelevant of course.

Reply:
Some people prefer the pattern above to show a default/exception pattern, show that the test is unlikely, or try to do the initialization in parallel with other stuff.
With the example below you lose track of what the default is, it looks more like a 50% probability of occurrence.

Reply:
@plbossart you can use `likely()` / `unlikely()` for that